App/Redux/PhotoViewingRedux.ts

```diff
@@ -0,0 +1,197 @@
+
+const actions = {
+  insertThread: createAction('INSERT_THREAD', resolve => {
+    return (thread: Thread) => resolve({ thread })
+  }),
+  addThreadRequest: createAction('ADD_THREAD_REQUEST', resolve => {
+    return (name: ThreadName, mnemonic?: Mnemonic) => resolve({ name, mnemonic })
+  }),
+  addThreadSuccess: createAction('ADD_THREAD_SUCCESS', resolve => {
+    return (thread: Thread) => resolve({ thread })
+  }),
+  addThreadError: createAction('ADD_THREAD_ERROR', resolve => {
+    return (error: any) => resolve({ error })
+  }),
+  removeThreadRequest: createAction('REMOVE_THREAD_REQUEST', resolve => {
+    return (id: ThreadId) => resolve({ id })
+  }),
+  removeThreadSuccess: createAction('REMOVE_THREAD_SUCCESS', resolve => {
+    return (id: ThreadId) => resolve({ id })
+  }),
+  removeThreadError: createAction('REMOVE_THREAD_ERROR', resolve => {
+    return (error: any) => resolve({ error })
+  }),
+  refreshThreadsRequest: createAction('REFRESH_THREADS_REQUEST'),
+  refreshThreadsError: createAction('REFRESH_THREADS_ERROR', resolve => {
+    return (error: any) => resolve({ error })
+  }),
+  refreshThreadRequest: createAction('REFRESH_THREAD_REQUEST', resolve => {
+    return (threadId: ThreadId) => resolve({ threadId })
+  }),
+  refreshThreadSuccess: createAction('REFRESH_THREAD_SUCCESS', resolve => {
+    return (threadId: ThreadId, photos: Photo[]) => resolve({ threadId, photos })
+  }),
+  refreshThreadError: createAction('REFRESH_THREAD_ERROR', resolve => {
+    return (threadId: ThreadId, error: any) => resolve({ threadId, error })
+  }),
+  viewWalletPhoto: createAction('VIEW_WALLET_PHOTO', resolve => {
+    return (photoId: PhotoId) => resolve({ photoId })
+  }),
+  viewThread: createAction('VIEW_THREAD', resolve => {
+    return (threadId: ThreadId) => resolve({ threadId })
+  }),
+  viewPhoto: createAction('VIEW_PHOTO', resolve => {
+    return (photoId: PhotoId) => resolve({ photoId })
+  }),
+  updateComment: createAction('UPDATE_COMMENT', resolve => {
+    return (comment: string) => resolve({ comment })
+  }),
+  addCommentRequest: createAction('ADD_COMMENT_REQUEST'),
+  addCommentSuccess: createAction('ADD_COMMENT_SUCCESS')
+}
+
+export type PhotoViewingAction = ActionType<typeof actions>
+
+export type ThreadData = {
+  readonly thread: Thread
+  readonly querying: boolean
+  readonly photos: ReadonlyArray<Photo>
+  readonly error?: Error
+}
+
+type ThreadMap = {
+  readonly [key: string]: ThreadData
+}
+
+type PhotoViewingState = {
+  readonly threads: ThreadMap,
+  readonly threadsError?: string
+  readonly addingThread?: {
+    readonly name: ThreadName,
+    readonly error?: string
+  }
+  readonly removingThread?: {
+    readonly id: ThreadId
+    readonly error?: string
+  }
+  readonly viewingWalletPhoto?: Photo,
+  readonly viewingThread?: Thread,
+  readonly viewingPhoto?: Photo,
+  readonly authoringComment?: string
```

Comment:
The above 4 properties is where we store info about what thread and photo is being viewed or operated on. Should probably move sharing related state in here too, but haven't yet.